src/libraries/System.Runtime.CompilerServices.Unsafe/tests/UnsafeTests.cs

```diff
@@ -931,6 +931,121 @@ public static void SkipInit_PreservesPrevious()
             Unsafe.SkipInit(out stringValue);
             Assert.Equal("25", stringValue);
         }
+
+        [Fact]
+        public static void IsNullRef_NotNull()
```

Comment:
I do not see any test that verifies that the value is actually NULL. If `NullRef` returned `(T*)7` and `IsNullRef` checked for it, I believe all tests would pass.

It may be good idea to:
- Verify that pinning `NullRef` using `fixed` produces null unmanaged pointer
- Verify that accessing `NullRef` produces NullReferenceException
- Verify `IsNullRef` called on byref created from unmanaged  pointer

On the other hand, I do not think it is necessary to test full matrix of primitive types. The implementation does not have any paths specific to given primitive type, so chances that one primitive types works but other primitive type does not are extremely low.

Reply:
That makes perfect sense, I'll add those new test cases!

As for the additional types being tests, I was following suit from the tests I saw for `Unsafe.SkipInit`, but yeah they're probably redundant in this case, I'll just leave one for each test method then 😊